committee-steering/sig-governance.md

```diff
@@ -0,0 +1,629 @@
+# SIG Governance Template
+
+## Goals
+
+The following document outlines a template governance structure for
+SIGs to follow.  Its goals are to provide the necessary structure
+for SIGs to self organize and self govern.
+
+This document is focused on:
+
+- Outlining organizational responsibilities
+- Outlining organizational roles
+- Outlining processes and tools
+
+Specific attention has been given to:
+
+- The role of technical leadership
+- The role of project leadership
+- The process for making technical decisions
+- The process for ensuring continued test health
+
+## How to use this template
+
+SIGs are diverse in nature, and may have different goals based
+on what areas of the project the SIG touches.  Accordingly SIGs will
+need to adapt this template to their specific needs.  This may mean
+removing or changing roles, responsibilities, and processes.  When
+diverging from this template, please help us understand why in the
+PR description so we can improve the process and iterate on this
+document.
+
+**Note:** There are 2 charter template versions:
+
+- a super-lite TL;DR version that only provides a abstract structure.
+- a detailed version that attempts to concretely define a high
+  degree of structure.
+
+SIGs may either choose to start with the TL;DR charter and add
+structure from the detailed version, or to start with the detailed
+version and remove structure that does not provide direct value.
+
+**Note:** Specific responsibilities towards the project are required
+for some SIGs and should be included in the charter - e.g. ensuring
+test health.
+
+FAQ:
+
+- [Why is it so long?](#why-is-it-so-long)
+- [Why so much structure?](#why-so-much-structure)
+- [Why use mailing lists for tracking roles?](#why-use-mailing-lists-for-tracking-roles)
+- [Why track membership as a special role?](#why-track-membership-as-a-special-role)
+- [How do the results from the survey fit in?](#how-do-the-results-from-the-survey-fit-in)
+
+---
+
+# SIG ~~Foo~~ Governance TL;DR Version
+
+## Roles
+
+- Project Leads
+  - Run operations and processes governing the SIG
+  - Outgoing Project Leads may propose new ones -
+  - Number: 2-3
+- Technical Leads
+  - Escalation for technical discussions
+  - Outgoing Technical Leads may propose new ones -
+  - Number: 2-7
+- Members
+  - Maintain health of the project
+  - Build new functionality
+
+## Processes
+
+- Agreeing upon decisions
+  - Send for lazy consensus - send to the googlegroup with RFC or KEP in
+    title
+  - Send LastCall email before accepting consensus
+  - Once accepted, decision is commitment requiring alternative proposal
+    to reverse
+- Test health
+  - Setup on call rotation (dashboard and tooling coming soon)
+  - Setup SLO for test health
```

Comment:
For some SIGs this is going to be a new level of commitment. I wonder if all SIGs have people who are willing to be "on call".

Reply:
That's a good question. 

According to the ["How to use this template"](https://github.com/kubernetes/community/pull/1650/files#diff-269660d0735677d8a36f8c4d75c1bdebR22) section, SIGs may change processes and responsibilities and adapt this template to their specific needs. The goal of having this template is to provide SIGs a starting point for creating a consistent way to communicate how a SIG functions. If a SIG does not have people willing to be "on call", they'd remove that item from their charter.

@pwittrock please correct me if I'm understanding or communicating anything incorrectly.

Does that help address your concern @mattfarina ?

Reply:
I would argue that at a minimum code and binary owners (be it subprojects or SIGs) must meet the following:

1. have confidence their code is working based on some automated signal
2. not release unless the signal shows it is working
3. make sure the signal shows that it is working when it is delivered as part of a coordinated release

I know some areas of the project do a great job of this already, and I know some don't.  For those doing a good job I don't think any action is required.  For areas that have self identified as needing help we should provide structure to help them.